File: Témata/05 NLRM s obecnou kovarianční maticí (heteroskedasticita)/Matlab/nu_lambda_post.m
function fpost = nu_lambda_post(nu_lambda,nu_lambda0,lambda)
N=length(lambda);


eta = 1/nu_lambda0+.5*sum(-log(lambda)+lambda);


fpost=N*nu_lambda/2*log(nu_lambda/2)-N*gammaln(nu_lambda/2)-eta*nu_lambda;

File: support/nu_lambda_post.py
import numpy as np
from math import gamma

def nu_lambda_post(nu_lam, nu_lam_0, lam):
    n = lam.shape[0]
    eta = 1 / nu_lam_0 + 0.5 * np.sum(lam - np.log(lam))
    return n * nu_lam / 2  * np.log(nu_lam) - n * np.log(gamma(nu_lam / 2)) - eta * nu_lam
